packages/material-ui-styles/src/useTheme.js

```diff
@@ -1,6 +1,8 @@
 import React from 'react';
+import warning from 'warning';
 import ThemeContext from './ThemeContext';
 
-export default function useTheme() {
+export default function useTheme(_) {
+  warning(!_, "Material-UI: useTheme() doesn't accept any argument.");
```

Comment:
Why do we need this?

Reply:
In case somebody do `useTheme(props)` or else.

Reply:
Yes I saw that. My question is why would we guard against this? The hook didn't have an argument previously, nor did `withTheme`. 

Different angle: Why do we only warn here at runtime that a nullary function was called with a single argument? I looks like that using a type checker like flow or typescript would be the first choice if somebody would be interested in warnings for wrong number of arguments.

Reply:
Ok, let's kill it.